src/App.tsx

```diff
@@ -1,14 +1,53 @@
 /* eslint-disable max-len */
-import React from 'react';
+import {
+  FC,
+  useState,
+  useEffect,
+  useCallback,
+  useMemo,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [filterBy, setFilterBy] = useState('all');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    getTodos()
+      .then(data => setTodos(data));
+  }, []);
+
+  const closeInfo = useCallback(() => {
+    setSelectedTodo(null);
+  }, []);
+
+  const visibleTodos = useMemo(() => (
+    todos.filter(todo => {
+      const queryFilter = todo.title.toLowerCase().includes(query.toLowerCase());
+
+      switch (filterBy) {
+        case 'active':
+          return queryFilter && todo.completed === false;
+
+        case 'completed':
```

Comment:
You can use enum for switch, but it works too

Reply:
agree, better use enum